databricks_cli/dbfs/cli.py

```diff
@@ -53,10 +55,20 @@ def ls_cli(api_client, l, absolute, dbfs_path): #  NOQA
         dbfs_path = dbfs_path[0]
     else:
         error_and_quit('ls can take a maximum of one path.')
-    files = DbfsApi(api_client).list_files(dbfs_path)
-    table = tabulate([f.to_row(is_long_form=l, is_absolute=absolute) for f in files],
-                     tablefmt='plain')
-    click.echo(table)
+    
+    def echo_path(files):
+        table = tabulate([f.to_row(is_long_form=l, is_absolute=absolute) for f in files],
+                         tablefmt='plain')
+        click.echo(table)
+    
+    def recursive_echo(this_dbfs_path):
+      files = DbfsApi(api_client).list_files(this_dbfs_path)
+      echo_path(files)
+      for f in files:
+          if f.is_dir:
+              recursive_echo(this_dbfs_path.join(f.basename))
```

Comment:
Do we have a rough estimation how much more requests this would generate? Do we have customers who already manually doing this and we are just giving them a shortcut?

Reply:
hi @stormwindy thanks for your review. i truly have no idea how many more requests it will generate; totally dependent on customer file structure. will be 1 extra request for each directory in the path and all of their subdirectories.

Reply:
I see. What are the use cases for this? Unless there is a crazy deep folder structure this shouldn't cause any problems. Overall I have no other concern.

Reply:
getting filepaths of all worker logs in a cluster: https://github.com/databricks/databricks-cli/issues/512